scripts/jira/cloner.py

```diff
@@ -0,0 +1,216 @@
+    data_dict['assignee'] = {'name': issue.fields.assignee.name}
+    data_dict['reporter'] = {'name': issue.fields.reporter.name}
+    data_dict['issuetype'] = {'name': issue.fields.issuetype.name}
+    data_dict['project'] = {'id': issue.fields.project.id}
+    data_dict['summary'] = issue.fields.summary
+    data_dict['description'] = issue.fields.description
+    data_dict['components'] = [{'id': x.name} for x in issue.fields.components]
+    data_dict['versions'] = [{'name': x.name} for x in issue.fields.versions]
+    data_dict['fixVersions'] = [{'name': x.name} for x in issue.fields.fixVersions]
+    # QA contact
+    if hasattr(issue.fields, 'customfield_12315948'):
+        data_dict['customfield_12315948'] = {'value': issue.fields.customfield_12315948.value}
+    data_dict['customfield_12320947'] = [{'value': x.value} for x in issue.fields.customfield_12320947]
+    # Target version
+    data_dict['customfield_12319940'] = [{'name': target}]
+
+    new_issue = connection.create_issue(data_dict)
+    connection.create_issue_link("Cloners", inwardIssue=new_issue.key, outwardIssue=issue.key)
+    connection.create_issue_link("Blocks", inwardIssue=issue.key, outwardIssue=new_issue.key)
+    return new_issue
+
+def check_clones(issue, connection, dry_run):
+    actions = []
+    try:
+        fix_versions = get_issue_fix_versions(issue)
+    except RuntimeError as e:
+        actions.append([JIRA_URL_PREFIX+issue.key, "", "Fix versions field empty. Please fill it."])
+        return actions
+
+    try:
+        target_version = get_issue_target_versions(issue)
+    except RuntimeError as e:
+        actions.append([JIRA_URL_PREFIX+issue.key, "", f"Target field problems: {e}. Please check."])
+        return actions
+
+    if not is_original_issue(issue):
+        return actions
+
+    if target_version != fix_versions[0]:
+        actions.append([JIRA_URL_PREFIX+issue.key, "", f"Target version does not target first Fix version. {target_version} != {fix_versions[0]}. Please check."])
+        return actions
+
+    fix_versions_missing = set(fix_versions[1:])
+    clones = get_clone_by_issues(issue, connection)
+    for clone in clones:
+        clone_fix_versions = []
+        try:
+            clone_fix_versions = get_issue_fix_versions(clone)
+        except RuntimeError as e:
+            if dry_run:
+                actions.append([JIRA_URL_PREFIX+clone.key, JIRA_URL_PREFIX+issue.key, f"Dry run. Skip set Fix versions."])
+            else:
+                try:
+                    set_fix_versions(clone, fix_versions, connection)
+                except RuntimeError as e:
+                    actions.append([JIRA_URL_PREFIX+clone.key, JIRA_URL_PREFIX+issue.key, f"Unable to update Fix versions: {e}"])
+                    continue
+                finally:
+                    actions.append([JIRA_URL_PREFIX+clone.key, JIRA_URL_PREFIX+issue.key, "Fix versions updated to match original"])
+
+        if clone_fix_versions != fix_versions:
```

Comment:
Does the order of the versions matter here? I don't know if Jira sorts the values or if we might fail to recognize "4.13.0, 4.12.z" being the same as "4.12.z, 4.13.0", for example.